### PR TITLE
API struct tagged to produce proper output when marshalled #217

### PR DESCRIPTION
**Pull request by kunalkushwaha**
Signed-off-by: Kunal Kushwaha <kunal.kushwaha@gmail.com>

### CODE REVIEW

File: api/types.go [hunk 1]
```diff
@@ -27,41 +27,41 @@ type endpointResource struct {
...
-	ParentUpdates     []endpointParentUpdate
-	UseDefaultSandbox bool
+	ContainerID       string                 `json:"container_id"`
+	HostName          string                 `json:"host_name"`
+	DomainName        string                 `json:"domain_name"`
+	HostsPath         string                 `json:"hosts_path"`
+	ResolvConfPath    string                 `json:"resolv_conf_path"`
+	DNS               []string               `json:"dns"`
+	ExtraHosts        []endpointExtraHost    `json:"extra_hosts"`
+	ParentUpdates     []endpointParentUpdate `json:"extra_hosts"`
```
Comment (aboch) [inline api/types.go]: typo. json tag should be "parent_updates"

**Comment by aboch** [comment]
You also need to change the code in client. Look at types.go there. Otherwise ci will fail.

**Comment by kunalkushwaha** [comment]
Thanks for review. Issue fixed.

**Comment by aboch** [comment]
Please have only one commit given the second fixes a breakage from the first one.

I want to hear from @dave-tucker if this addresses his request in issue #217.
Specifically, if we go with the json tagging route, then shouldn't we tag the backend data structures and get rid of the response body structures in api/types.go, which were added for lack of tagging ?

**Comment by aboch** [comment]
Actually reading again issue #217, it looks like the issue is only about avoiding camelCase and capitalized fields in the json string... Then this is good. Only one more comment on the diffs and please have only one commit. Thanks.

File: api/types.go [hunk 2]
```diff
@@ -8,17 +8,17 @@ import "github.com/docker/libnetwork/types"
...
 // networkResource is the body of the "get network" http response message
 type networkResource struct {
-	Name      string
-	ID        string
-	Type      string
-	Endpoints []*endpointResource
+	Name      string              `json:"name"`
+	ID        string              `json:"id"`
+	Type      string              `json:"type"`
+	Endpoints []*endpointResource `json:"Endpoints"`
```
Comment (aboch) [inline api/types.go]: Can you change `Endpoints` to `endpoints`

**Comment by kunalkushwaha** [comment]
Closing this. Created a new PR,

**Pull request closed without merging**